### ISSUE TITLE
[scan] A prior campaign takes over engin-protein's entire conformal calibration split — 25/25 variants in the shipped example — so the interval is calibrated on a different protein and the EI incumbent becomes the prior's maximum

### ISSUE THREAD

**Issue by dcbaugher** [enginbio/engin-suite#413]
**Lens:** Depth (rotation idx 0) · **Scanned:** 2026-09-16

## What

`CampaignPlanner._seed_variants` builds the training pool as *current campaign first, prior
campaigns appended*:

```python
pooled = list(campaign.measured())
for prior in self.prior_campaigns:
    if prior.length != campaign.length:
        continue
    pooled.extend(prior.measured())
```

`LowNCopilot.fit` then takes the **tail** of that pool as the conformal calibration set:

```python
n_cal = max(2, int(round(self.cal_fraction * len(variants))))   # cal_fraction = 0.3
self.model.fit(variants[:-n_cal]).calibrate(variants[-n_cal:], level=level)
self._best_observed = float(max(v.fitness for v in variants))
```

Because priors are appended last and are typically larger than the seed campaign, the tail is
*entirely* prior-campaign variants. In `examples/run_planner.py`'s own configuration — 24-variant
seed campaign, 60-variant prior — `n_cal = 25` and **all 25 come from the prior**, at every
similarity setting, deterministically. Round 1 is worse than a random split would be: it is a
split that cannot contain a single target-campaign variant.

Two consequences ride on the same line:

1. **The split-conformal guarantee is over the calibration distribution.** With a prior supplied,
   that distribution is a *different landscape*. `ScoredDesign.lower`/`upper` are returned as the
   calibrated interval regardless, and nothing in the return value records which protein the
   multiplier came from.
2. **`_best_observed` is the max over the pooled set**, so the EI incumbent is the prior's best
   variant. `Variant`'s validator instructs callers to "Normalize per-assay before building a
   Campaign", and min-max normalization puts every real prior campaign's maximum at exactly 1.0 —
   so under the schema's own instruction the incumbent pins at the top of the scale for any
   campaign with a prior.

## Why it matters

The measured symptom is **not** under-coverage — it is the opposite, and that is what makes it
quiet. The interval goes conservative and then vacuous while the coverage number stays green.

Measured on the bundled synthetic landscape (`make_landscape(epistasis=0.5, seed=0)`),
60-variant seed campaign, 60-variant prior, nominal level 0.90, coverage on 400 held-out
**target** variants, 8 seeds. `top8 overlap` is the recommended batch against the same copilot
fitted on the target campaign alone:

| prior | n_pool | n_cal | % of cal from prior | coverage on target | mean width | incumbent | top-8 overlap |
|---|---|---|---|---|---|---|---|
| none | 60 | 18 | 0% | 0.909 | 0.779 | 0.947 | 8.0/8 |
| `related(similarity=0.9)` | 120 | 36 | **100%** | 0.952 | 0.729 | 0.968 | **2.9/8** |
| `related(similarity=0.0)` | 120 | 36 | **100%** | 0.976 | **1.008** | 1.000 | **1.1/8** |
| `related(similarity=0.0)`, min-max renormalized | 120 | 36 | **100%** | 0.985 | **1.188** | 1.000 | **1.0/8** |

Fitness is `[0, 1]` by schema, so a mean width of 1.008 is an interval **wider than the entire
assay range** — it covers because it cannot fail to. A coverage test would pass on every row.

The overlap column is the part a user pays for: with a *related* prior — the favourable case
cross-project transfer is sold on — the recommended batch shares 2.9 of 8 variants with the batch
the same model recommends without the prior. That is a different round of assays, driven by a
calibration set and an incumbent that both came from another protein.

This is `engin-protein`'s only cross-project mechanism, and `planner.py` calls it the thing that
differentiates the package: *"What can differentiate is biology-specific priors... Cross-project
priors are the claimed moat."*

Secondary, same function: the only guard on pooling is `prior.length != campaign.length`. Two
unrelated proteins of the same length are pooled positionally with no other check.

## Evidence

- `packages/engin-protein/src/engin_protein/planner.py:41-53` — `_seed_variants` appends priors
  after the current campaign; the only filter is sequence length.
- `packages/engin-protein/src/engin_protein/lown.py:62-71` — `fit` takes `variants[-n_cal:]` as
  the calibration split and `max(v.fitness for v in variants)` as the incumbent, both over the
  pooled list.
- `packages/engin-protein/src/engin_protein/model.py:262-291` — `calibrate`'s stated precondition
  is only *"variants the model was not fit on"*. Exchangeability with the target campaign is not
  stated and is what the pooling path breaks.
- `packages/engin-protein/src/engin_protein/schema.py:47-55` — `Variant` rejects fitness outside
  `[0,1]` with *"Normalize per-assay before building a Campaign"*, which is what forces a real
  prior's maximum to 1.0.
- `packages/engin-protein/src/engin_protein/landscape.py:172-200` — `related()` blends the
  fields, then **recalibrates `_lo`/`_hi` on its own probe sample**. Every landscape is normalized
  to `[0,1]` independently, so `similarity=0.0` is scale-matched to the target by construction.
  The honest control the project is proud of controls for landscape *shape* and not for *scale*.
- `packages/engin-protein/build-log.md:62-65` — *"Transfer is not validated... `transfer_benefit`
  tests assert structure only, never lift."* The declared gap is about lift; nothing here or in
  `examples/run_planner.py` mentions the calibration split.
- Siska et al., *A Guide to Bayesian Optimization in Bioprocess Engineering*,
  [arXiv:2508.10642](https://arxiv.org/abs/2508.10642) / Biotechnol. Bioeng. 2026,
  [10.1002/bit.70129](https://doi.org/10.1002/bit.70129) — §4.1 defines batch effects as
  *"systematic biases between experimental batches or samples, caused by factors unrelated to the
  design variables"* and recommends modelling the batch term explicitly. Adjacent rather than
  on-point: it is about run-day effects inside one campaign, not about conformal calibration
  across campaigns. It is cited for the premise that a cross-campaign offset is the expected
  condition, not the exceptional one.

Reproduction (8 seeds, ~2 min):

```python
from engin_protein import make_landscape, CampaignPlanner, Campaign
from engin_protein.lown import LowNCopilot

target = make_landscape(epistasis=0.5, seed=0)
seed_c = target.sample_campaign(24, seed=30, campaign_id="target")
prior  = target.related(similarity=0.0, seed=11).sample_campaign(60, seed=9, campaign_id="prior")

train = CampaignPlanner(prior_campaigns=[prior])._seed_variants(seed_c)
n_cal = max(2, int(round(0.3 * len(train))))
cal   = train[-n_cal:]
print(sum(v.variant_id.startswith("prior") for v in cal), "/", n_cal)   # -> 25 / 25

cop = LowNCopilot().fit(Campaign(campaign_id="r0", variants=train))
print(cop._best_observed)                                              # -> the prior's max
```

## What it would take

The smallest version that settles it is a **stratified split**: draw the calibration set from the
target campaign only, and use pooled priors for fitting alone. That is a slice change in
`LowNCopilot.fit` plus a marker on `Campaign` (or on `Variant`) saying which campaign a variant
came from, since `fit` currently receives one flat `Campaign` and cannot tell. `_best_observed`
should come from the same target-only subset for the same reason `stopping.posterior_incumbent`
exists — it is a threshold, not a ranking offset.

Note the split is not free: pooling is *why* `n_cal` rises from 18 to 36, and more calibration
points is exactly what `level_for_split` and #197 want in the low-N regime. A target-only split
gives back the honest interval and takes away the points. Which is the better trade is the
question the change should answer, and it is answerable on the bundled landscape today.

The fuller answer is the hierarchical prior `_seed_variants` already names as M1+, or a Mondrian
split with campaign as the taxonomy. Both are larger than this.

Separately, and cheaply: `FitnessLandscape.related()` cannot currently produce a scale-shifted
prior, so no experiment in this repository can exhibit the failure a real cross-assay prior would
cause. A `scale=` / `offset=` argument would make the existing honest control able to see it.

## Decisions touched

`D12` — implements. A tier-1 result the tier-1 row does not mention, in the direction `D12` cares
about: the reported coverage is fine and the thing it is reported on is not. `D14` neutral. `D9`
neutral — the fix is a slice, not a dependency. Contradicts no standing decision.

## Why this might be wrong

**The strongest objection is that transfer is already declared unvalidated**, loudly, in two
places — `build-log.md` and the paragraph `run_planner.py` prints at the end. A reader could say
this issue re-files a known gap. I do not think it survives: what is declared unvalidated is the
*lift*, measured as best-found fitness. The calibration split is a different quantity, it is
deterministic rather than statistical, and `ScoredDesign` hands a user `lower`/`upper` and
`prob_above_threshold` from it with no qualification. On a package whose stated deliverable is the
label on the interval, "the label is computed on a different protein" is not covered by "the lift
is unproven".

**The second objection is that I am calling conservative coverage a defect.** Coverage rises —
0.909 → 0.952 → 0.976 — and too-wide is the safe direction, which `build-log.md` says explicitly
about this package. That is fair as far as it goes, and it is why the table above leads with width
rather than coverage. The claim is not that the interval under-covers; it is that the guarantee
attaches to the prior campaign's residual distribution rather than the target's, and that the
observable price is an interval wider than the whole assay range. A conservative interval that is
conservative *for an unknown reason* is not the same object as one that is conservative because
n_cal is small, and only the second is what #197 and `level_for_split` are managing.

**The third is that the magnitude is understated and I cannot fix that here.** Both landscapes are
normalized to `[0,1]` by `related()`, so even the renormalized row is a mild version of what a
real prior from a different assay would do. That cuts both ways: it means the numbers above are a
floor, and it also means I have no measurement of the real-campaign magnitude and am not claiming
one. The 1.188 row is the closest available, and it is still synthetic.

**Fourth, the top-8 overlap column is not by itself evidence of harm.** A different batch is not a
worse batch, and `build-log.md` reports a positive mean lift even at similarity 0.0 (+0.023). The
overlap is here to show the defect reaches the recommendation and not only the interval; it is not
a claim that the recommendation degrades. Establishing *that* needs the seed counts the build log
already says this question deserves.

**Fifth**, one could argue the ordering in `_seed_variants` is incidental and a caller could shuffle
before fitting. They could — but no caller does, `run()` does not, and a fix that depends on every
caller knowing to shuffle is the kind of thing `D14` says a library should not require.
